### PR TITLE
Unpin dask version and unset dask environment variables

### PR DESCRIPTION
**Pull request by jnke2016**
This PR unpins the dask version and unsets timeout related environment variables.

### CODE REVIEW

File: conda/environments/all_cuda-120_arch-x86_64.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ dependencies:
 - cupy>=12.0.0
 - cxx-compiler
 - cython>=3.0.0
-- dask-core==2023.9.2
+- dask-core>=2023.7.1
 - dask-cuda==23.12.*
 - dask-cudf==23.12.*
 - dask>=2023.7.1
```
Comment (naimnv) [inline conda/environments/all_cuda-120_arch-x86_64.yaml:26]: Same here

**Comment by naimnv** [comment]
LGTM

**Comment by jnke2016** [comment]
It looks like CI is still testing this PR with the pin dask distributed version
`distributed 2023.9.2  pyhd8ed1ab_0 `

**Comment by jnke2016** [comment]
Already done in this [PR](https://github.com/rapidsai/cugraph/pull/3953)

**Pull request closed without merging**